pkg/libs/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -154,22 +174,30 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 	return utilerrors.NewAggregate(errors)
 }
 
-func (c *endpointAccessibleController) buildTLSClient() (*http.Client, error) {
+func (c *endpointAccessibleController) buildTLSClient() (*http.Client, bool, error) {
```

Comment:
I don't think that this is a good change overall. You expect `buildTLSClient` to return either a working client or an error.
You don't expect it to return a state message.

Another option would be to call the `c.GetTLSConfigFn()` on the stack above and hand that object into `buildTLSClient` and add a TLS change check method.

```go
	tlsConfig, err := c.buildTLSConfigFn()
	if err != nil { /* ... */ }

	tlsChanged, err := c.hasTLSChanged(tlsConfig)
	if err != nil {
		return err
	}

	isPastTimeForCheck := time.Since(c.lastCheckTime) > c.maxCheckLatency
	if !endpointsChanged && !tlsChanged && !isPastTimeForCheck {
		return nil
	}


	client, err := c.buildTLSClient(tlsConfig)
	if err != nil {
		return err
	}

	c.lastCheckTime = time.Now()
	c.lastEndpoints = newEndpoints
```

Reply:
That's a nice idea :+1:

Reply:
Thank you for your patience and appreciation.